### PR TITLE
Add: sql_view - create SQL views on the database

### PR DESCRIPTION
**Pull request by guewen**
This addon allows to create SQL views on the database.  It also features
a simple CSV export of the views to check their result.

### CODE REVIEW

File: sql_view/models/sql_view.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    def _complete_from_sql_name(self, cr, uid, sql_name, context=None):
+        return SQL_VIEW_PREFIX + (sql_name or '')
+
+    def _compute_complete_sql_name(self, cr, uid, ids, name, args,
+                                   context=None):
+        res = {}
+        for sql_view in self.browse(cr, uid, ids, context=context):
+            res[sql_view.id] = self._complete_from_sql_name(cr, uid,
+                                                            sql_view.sql_name,
+                                                            context=context)
```
Comment (sebalix) [inline sql_view/models/sql_view.py:48]: I am not a fan of this args' indentation, as it forces to re-indent all the lines if we change the length of the first one and make the reading harder. I always put the arguments on the second line if there is not enough space, like this:

``` python
            res[sql_view.id] = self._complete_from_sql_name(
                cr, uid, sql_view.sql_name, context=context)
```

But it is just a matter of taste.

File: sql_view/wizards/sql_view_csv_preview.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        form = self.browse(cr, uid, ids[0], context=context)
+        sql_view = self.pool['sql.view'].browse(cr, uid, sql_view_ids[0],
+                                                context=context)
+        query = self._query(cr, uid, form, sql_view, context=context)
+        cr.execute(query)
+        headers = [desc[0] for desc in cr.description]
+        records = cr.fetchall()
+        filedata = StringIO()
+        try:
+            writer = unicodecsv.writer(filedata, encoding='utf-8')
```
Comment (sebalix) [inline sql_view/wizards/sql_view_csv_preview.py:69]: Is it possible to use the `csv_internal_sep` server option as the delimiter to be consistent with other exports?

**Comment by sebalix** [comment]
Tested, working fine.
Great module to maintain SQL views :+1:

File: sql_view/wizards/sql_view_csv_preview.py (hunk 2 above)
Comment (guewen) [inline sql_view/wizards/sql_view_csv_preview.py:69]: from a quick grep, this option does not seem to be used?
I can even find in a test (openerp/tests/addons/test_impex/tests/test_import.py)

> ```
>  [[ # FIXME: hardcoded comma, import uses config.csv_internal_sep
>            # resolution: remove configurable csv_internal_sep
> ```

Comment (sebalix) [inline sql_view/wizards/sql_view_csv_preview.py:69]: No problem then :) I thought this option was used, thanks for the tip.

**Comment by legalsylvain** [comment]
I gave a quick overview of this PR. It seems that as you said, there are some common functionnalities with #110.
@guewen : What do you think about that point, avoiding duplicated features ?

**Comment by guewen** [comment]
@legalsylvain about functionalities, they have nothing that they share. This module handles views, when #110 handle materialized views. At first sight, I thought that both could be mixed but then I noticed that from a code point of view they have very few things they could share, so I don't see much benefits to mix both modules.

**Comment by max3903** [comment]
:+1:

**Pull request merged**